tag-system/src/main/java/org/togetherjava/tjbot/tag/TagSystem.java

```diff
@@ -0,0 +1,66 @@
+package org.togetherjava.tjbot.tag;
+
+import org.jooq.Result;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.Tags;
+import org.togetherjava.tjbot.db.generated.tables.records.TagsRecord;
+
+import java.util.*;
+
+public class TagSystem {
+    private final Database database;
+
+    public TagSystem(Database database) {
+        this.database = database;
+    }
+
+    public boolean exists(String tag) {
+        return database.readTransaction(ctx -> {
+            return Optional
+                .ofNullable(ctx.selectFrom(Tags.TAGS).where(Tags.TAGS.ID.eq(tag)).fetchOne())
+                .isPresent();
+        });
+    }
+
+    public void delete(String tag) {
+        database.writeTransaction(ctx -> {
+            ctx.deleteFrom(Tags.TAGS).where(Tags.TAGS.ID.eq(tag)).execute();
+        });
+    }
+
+    public void put(String tag, String content) {
+        database.writeTransaction(ctx -> {
+            TagsRecord tagsRecord = ctx.newRecord(Tags.TAGS).setId(tag).setContent(content);
+
+            if (tagsRecord.update() == 0) {
```

Comment:
Most databases provide a facility of upserting records. That would make it an atomic statement as well as a single database call. Putting links below, I am assuming jooq merge will change it to sqlite upsert.

Ref:
https://www.jooq.org/doc/3.1/manual/sql-building/sql-statements/merge-statement/
https://www.sqlite.org/lang_UPSERT.html

Reply:
Ignored 😢

Reply:
Yea, because I had no clue what you were talking about / suggesting.

Reply:
Try something like this:

```suggestion
ctx.insertInto(Tags.TAGS, Tags.TAGS.ID, Tags.TAGS.CONTENT)
      .values(tag, content)
      .onDuplicateKeyUpdate()
      .set(Tags.TAGS.CONTENT, content)
      .execute();
```

Reply:
done